File: 1431. Kids With the Greatest Number of Candies/1431.go
package leetcode

func kidsWithCandies(candies []int, extraCandies int) []bool {
	maxNum := getMax(candies)
	res := make([]bool, 0, len(candies))
	for _,v := range candies{
		if v+extraCandies>=maxNum{
			res = append(res, true)
		}else{
			res = append(res, false)
		}
	}
	return res
}

func getMax(candies []int) int{
	res := candies[0]
	for _,v := range candies{
		if v > res{
			res = v
		}
	}
	return res
}

File: 1431. Kids With the Greatest Number of Candies/1431.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
bool* kidsWithCandies(int* candies, int candiesSize, int extraCandies, int* returnSize){
    *returnSize = candiesSize;
    bool* res = (bool*)malloc(candiesSize*sizeof(bool));
    memset(res, 0, candiesSize*sizeof(bool));
    int maxNum = getMax(candies,  candiesSize);
    for(int i=0; i<candiesSize; i++){
        if(*(candies+i)+extraCandies >= maxNum){
            *(res+i) = true;
        }else{
            *(res+i) = false;
        }
    }
    return res;
}

int getMax(int* candies, int candiesSize){
    int res = *(candies);
    for(int i=1; i<candiesSize; i++){
        if(*(candies+i) > res){
            res = *(candies+i);
        }
    }
    return res;
}